src/handlers/int-test-support/helpers/mock-data/csv/objectsToCsv.ts

```diff
@@ -0,0 +1,71 @@
+interface ObjectsToCSVOptions {
+  filterKeys?: string[]; // a list of keys to be filtered out of the output csv
+  renameKeys?: Map<string, string>; // a dictionary of keys to be renamed
+}
+type ObjectsToCSV = (
+  objects: Array<Record<string, string | number>>,
+  options?: ObjectsToCSVOptions
+) => string;
+
+export const objectsToCSV: ObjectsToCSV = (objects, options) => {
+  // Filter objects with given keys
+  if (options?.filterKeys !== undefined) {
+    filterKeys(objects, options?.filterKeys);
+  }
+
+  // Set up header row
+  const csvString = [];
+  const headerRow = [
+    ...new Set(objects.map((object) => Object.keys(object)).flat()),
+  ];
+  console.log(headerRow);
+
+  // Rename header row items with given dictionary
+  if (options?.renameKeys !== undefined) {
+    csvString.push(renameHeaderKeys(headerRow, options?.renameKeys));
+  } else {
+    csvString.push(headerRow);
+  }
+
+  // Build CSV String
+  for (let i = 0; i < objects.length; i++) {
+    const row = [];
+    for (let j = 0; j < headerRow.length; j++) {
+      row.push(objects[i][headerRow[j]]);
+    }
+    csvString.push(row);
+  }
+  return csvString.map((e) => e.join(",")).join("\n");
```

Comment:
```suggestion
  for (const object of objects) {
    const row = [];
    for (const header of headerRow) {
      csvString += `${object[header]},`;
    }
    csvString += '\n';
  }
  return csvString;
```

Reply:
note you'll need to concat the header row onto the string in a similar way too if you do this. TBH it was tricky of you to call an array "csvString"

Reply:
I need to get away from the simple for loops I am attached to so I will use your for...of suggestion but I'd like to keep working with it as an array until the end - I personally find it easier to understand and work with - but I will change the variable name from csvString to csvData.